### DISCUSSION TITLE
Grafana - How to keep secrets out of view (ENV VARS?)

### DISCUSSION THREAD

**Discussion by optimadba** [prometheus-operator/kube-prometheus#1629] [General]
Hi,

For starters, I'm not an expert on k8s, jsonnet etc.  I am trying to deploy the kube-prometheus stack after adapting my example.jsonnet to reflect my environment and components I need.  It has been a rather steep learning curve, and a rocky road at best, but we're 99% there.

We have Grafana configured to use our AzureAD for authentication, and here we run into a challenge.  AzureAD works like a charm, but the secrets (client_id, client_secret) are in plain text in the example.jsonnet file. We make use of the azure keyvault in our environment and we can make the secrets containing the said values available in the namespace where kube-prometheus is deployed to.  

Where we've (i've)run into a brick wall is to get the secrets included into the Grafana configuration.  I have found the discussion #https://github.com/prometheus-operator/kube-prometheus/discussions/631 and have tried to apply that, but no joy thusfar.
[snippet.txt](https://github.com/prometheus-operator/kube-prometheus/files/8039719/snippet.txt)

Looking forward to some valuable advice,

Thomas

**Comment by mvvveen** [comment]
Same issue here
